src/context/WalletProvider/KeepKey/components/RecoveryInvalid.tsx

```diff
@@ -0,0 +1,56 @@
+import { NotAllowedIcon } from '@chakra-ui/icons'
+import { Button, ModalBody, useToast } from '@chakra-ui/react'
+import { RecoverDevice } from '@shapeshiftoss/hdwallet-core'
+import { useState } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { Text } from 'components/Text'
+import { useWallet } from 'hooks/useWallet/useWallet'
+
+import { parseIntToEntropy } from './Label'
+
+export const KeepKeyRecoveryInvalid = () => {
+  const [loading, setLoading] = useState(false)
+  const {
+    setDeviceState,
+    state: {
+      deviceState: { recoverWithPassphrase, recoveryEntropy },
+      wallet,
+    },
+  } = useWallet()
+  const translate = useTranslate()
+  const toast = useToast()
+
+  const handleRetryPress = async () => {
```

Comment:
I moved it to a hook, because the error handling is the same on both component too and it avoids sending a lot of parameters to an helper function

Hope the naming is fine